Shared/Connect-M365.ps1

```diff
@@ -0,0 +1,159 @@
+        }
+    }
+
+    #Validate EXO V2 is connected or try to connect
+    $connection = $null
+    $connection = Get-ConnectionInformation -ErrorAction SilentlyContinue
+    if ($null -eq $connection) {
+        Write-Host "Please use Global administrator credentials" -ForegroundColor Yellow
+        if ($Force -or $PSCmdlet.ShouldContinue("Do you want to connect?", "We need a ExchangeOnlineManagement connection")) {
+            Connect-ExchangeOnline -ErrorAction SilentlyContinue
+            $connection = Get-ConnectionInformation -ErrorAction SilentlyContinue
+            if ($null -eq $connection) {
+                Write-Host "Connection could not be established" -ForegroundColor Red
+                Write-Host "We cannot continue without ExchangeOnlineManagement Powershell session" -ForegroundColor Red
+                return $false
+            } else {
+                Write-Host "Connected to EXO V2"
+                Write-Host "Session details"
+                Write-Host "Tenant Id: $($connection.TenantId)"
+                Write-Host "User: $($connection.UserPrincipalName)"
+                return $true
+            }
+        } else {
+            Write-Host "We cannot continue without ExchangeOnlineManagement Powershell session" -ForegroundColor Red
+            return $false
+        }
+    } elseif ($connection.count -eq 1) {
+        Write-Host "Connected to EXO V2"
+        Write-Host "Session details"
+        Write-Host "Tenant Id: $($connection.TenantId)"
+        Write-Host "User: $($connection.UserPrincipalName)"
+        return $true
+    } else {
+        Write-Host "You have more than one EXO sessions please use just one session" -ForegroundColor Red
+        return $false
+    }
+}
+
+function Test-AADConnection {
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    [OutputType([bool])]
+    param (
+        [Switch]$Force
+    )
+
+    #Validate AzureAD is installed
+    if ((Get-Module -ListAvailable | Where-Object { $_.Name -like "AzureAD" }).count -ge 1) {
+        Write-Host "AzureAD Powershell Module installed"
+    } else {
+        if ($Force -or $PSCmdlet.ShouldContinue("Do you want to install the module?", "AzureAD Powershell Module not installed")) {
+            Install-Module -Name AzureAD -Repository PSGallery -AllowClobber -Force -ErrorAction SilentlyContinue -Scope CurrentUser
+            if ((Get-Module -ListAvailable | Where-Object { $_.Name -like "AzureAD" }).count -ge 1) {
+                Write-Host "AzureAD Powershell Module installed"
+            } else {
+                Write-Host "AzureAD Powershell Module installation failed" -ForegroundColor Red
+                return $false
+            }
+        } else {
+            Write-Host "We cannot continue without AzureAD Powershell module" -ForegroundColor Red
+            return $false
+        }
+    }
+
+    #Validate AzureAD is loaded
+    if ((Get-Module | Where-Object { $_.Name -like "AzureAD" }).count -ge 1) {
```

Comment:
The AzureAD PowerShell module is deprecated. We should use PowerShell Graph SDK instead or using the existing shared Graph functions to acquire a token and run Graph API calls.

More information: https://techcommunity.microsoft.com/t5/microsoft-entra-blog/important-update-deprecation-of-azure-ad-powershell-and-msonline/ba-p/4094536

Reply:
Good point, we should maybe also require a min version of the modules?